### PR TITLE
[nextion] add `skip_connection_handshake` attribute

### PR DESCRIPTION
**Pull request by edwardtfn**
## Description:

**Related issue (if applicable):** fixes esphome/issues#2872

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6905

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by edwardtfn** [comment]
This is not stale.

File: components/display/nextion.rst [hunk 1]
```diff
@@ -60,6 +60,9 @@ Configuration variables:
 - **wake_up_page** (*Optional*, int): Sets the page to display after waking up
 - **auto_wake_on_touch** (*Optional*, boolean): Sets if Nextion should auto-wake from sleep when touch press occurs.
 - **exit_reparse_on_start** (*Optional*, boolean): Request the Nextion exit Active Reparse Mode before setup of the display. Defaults to ``false``.
+- **skip_connection_handshake** (*Optional*, boolean): Sets whether the Nextion display should skip the connection handshake process.
```
Comment (kbx81) [inline components/display/nextion.rst]: I think I would rephrase this so it's more correct -- since the _Nextion_ doesn't skip the handshake, but rather _ESPHome_ skips its handshake. haha

```suggestion
- **skip_connection_handshake** (*Optional*, boolean): Sets whether the initial display connection handshake process is skipped.
```

**Review by kbx81 — APPROVED** [review]
